src/firefly/js/ui/SimpleComponent.jsx

```diff
@@ -34,3 +34,36 @@ export class SimpleComponent extends PureComponent {
         }
     }
 }
+
+
+
+/**
+ * A replacement for SimpleComponent.
+ * This function make use of useState and useEffect to
+ * trigger a re-render of functional components when the value in the store changes.
+ *
+ * @param stateGetters  one or more functions returning a state
+ * @returns {Object[]}  an array of state's value in the order of the given stateGetters
+ */
+export function useStoreConnector(...stateGetters) {
+
+    const rval = [];
+    const setters = stateGetters.map((getter) => {
+        const [val, setter] = useState(getter());
+        rval.push(val);
+        return [getter, setter];
+    });
+
+    const isMounted = true;
+    useEffect(() => {
+        return flux.addListener(() => {
+            if (isMounted) {
+                setters.forEach(([getter, setter]) => {
+                    setter(getter());
```

Comment:
don't you want to check for equality here like you use to?

Reply:
It's not needed since React will do an Object.is() on the value and only trigger render when it return false.